include/rmm/mr/device/cuda_async_memory_resource.hpp

```diff
@@ -44,6 +44,17 @@ namespace rmm::mr {
  */
 class cuda_async_memory_resource final : public device_memory_resource {
  public:
+  /**
+   * Flags for specifying particular handle types
```

Comment:
Probably worth mentioning that these values are exact copies of cudaMemAllocationHandleType, why we need to define our own type for this when it's a carbon copy, and adding a link where the values can be found for future reference (I found it here, but this link is pretty ugly and I don't know if it's persistent
https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1gabde707dfb8a602b917e0b177f77f365)

Reply:
I deleted my comment suggesting this when I failed to find a pretty URL for the reference. 😢 An explanatory comment, even without the link, would still be helpful.

Reply:
If we're worried about the persistence of the link (or the appearance, although I care a lot less about that and it's in a comment anyway) we can always just link to the page and tell users what keyword to search for e.g. "see the cudaMemAllocationHandleType docs at https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html".